assignment/core/src/main/java/com/assignment/core/service/impl/NodeServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package com.assignment.core.service.impl;
+
+import com.assignment.core.services.NodeService;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jcr.Node;
+
+/**
+ * Implementation of the NodeService interface.
+ * Provides methods for processing node operations, including saving and deleting nodes.
+ */
+@Component(service = NodeService.class)
+public class NodeServiceImpl implements NodeService {
+
+    private static final Logger logger = LoggerFactory.getLogger(NodeServiceImpl.class);
+
+    private static final String USERGEN_PATH = "/content/usergenerated/session20";
+
+    /**
+     * Processes the save or delete operation based on the provided flag.
+     * Creates or deletes nodes under the specified path and commits the changes.
+
+     */
+    @Override
+    public void processSaveOrDelete(ResourceResolver resolver, String saveFlag, long matchCount) throws Exception {
+        boolean saveNode = Boolean.parseBoolean(saveFlag);
+        Resource userGeneratedRes = resolver.getResource("/content/usergenerated");
+
+        if (userGeneratedRes != null) {
+            Node userGeneratedNode = userGeneratedRes.adaptTo(Node.class);
+
+            if (userGeneratedNode != null) {
+                logger.info("Processing user-generated nodes...");
+
+                if (saveNode) {
```

Comment:
could have made 2 separate methods to save/delete (like deleteNode()) and can be called from processSaveOrDelete() based on your flag. so that these methods can be utilized by other business logic in the future